### PR TITLE
refactor: Move code from main.go to other files

### PR DESCRIPTION
**Pull request by jirihnidek**
* Card ID: CCT-950
* Initial step of refactoring main.go.
* Code moved from main.go to other files
* Introduced new files for connect and disconnect commands
* Added several other files for interactive mode, logging
* Some code was moved to existing files
* Unused code was removed.

### CODE REVIEW

**Review by subpop — CHANGES_REQUESTED** [review]
I have a concern with the change caused by this PR. While I see the benefit to reorganizing the files, this PR will create conflicts that need to be resolved in other PRs when they rebase onto these changes. This project is under fairly active development by a few different authors, and I would like to avoid putting unnecessary conflict resolution into those other PRs, if possible. Could we be more methodical about the refactor? Perhaps we could move the functionality of subcommands one commit at a time? Or first mark these "action" functions as deprecated, build new subcommand data structures that call back into these original "action" functions and then remove them?

I do something similar to this in [xrhidgen](https://github.com/subpop/xrhidgen). Each subcommand is declared as a global variable (for instance, [`associateCommand`](https://github.com/subpop/xrhidgen/blob/667c0343426453a6e3427f0b8e2d0cc359a38de5/cmd/xrhidgen/associate.go#L34)). Because that variable is a `Command` instance, it can be listed directly in the top level list of subcommands (as seen [here in main.go](https://github.com/subpop/xrhidgen/blob/667c0343426453a6e3427f0b8e2d0cc359a38de5/cmd/xrhidgen/main.go#L60C1-L66C5)).

Perhaps we can do something like that? There is no scenario I can think of that won't force some sort of conflict resolution upon the other outstanding PRs, but I'd like to make sure we do what we can to reduce the amount of conflict resolution we put upon the other PRs.

File: util.go [hunk 1]
```diff
@@ -70,71 +68,6 @@ func ConfigPath() (string, error) {
 	return filePath, nil
 }
 
-// GuessAPIURL gets the API server URL based on, insights-client.conf
-// and rhsm.conf. This URL may differ from prod, stage and Satellite
-func GuessAPIURL() (string, error) {
```
[the remark this answers is not in the document]
Comment (jirihnidek) [reply util.go:75]: This function was not used. I also removed two other unused functions. It seems that CI linter checks only changed code. It is reason, why I was able to see these issues, when I was moving lot of code.

**Comment by jirihnidek** [comment]
> I have a concern with the change caused by this PR. While I see the benefit to reorganizing the files, this PR will create conflicts that need to be resolved in other PRs when they rebase onto these changes. This project is under fairly active development by a few different authors, and I would like to avoid putting unnecessary conflict resolution into those other PRs, if possible. Could we be more methodical about the refactor? Perhaps we could move the functionality of subcommands one commit at a time? Or first mark these "action" functions as deprecated, build new subcommand data structures that call back into these original "action" functions and then remove them?

There is no open PR that could IMHO have conflict with this PR, because other opened PRs are related to build system & `rhc.spec` file or integration tests. Only my PR #46 could have some conflicts, but it will be easy to resolve. I think that this is great time to do this refactoring in one PR.

File: util.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (subpop) [reply util.go:75]: I agree. It's easier to trace through history when changes like this are made in separate commits.

**Comment by jirihnidek** [comment]
@subpop @m-horky I updated the PR as requested.

**Review by m-horky — APPROVED** [review]

**Pull request merged**